### PR TITLE
Dynamic search box updates

### PR DESCRIPTION
**Pull request by dctoon**
Remove the wildcard, and computed the query server side.
Fixed loading issue when display functions fail.
Fixed MultiSelect CSS issue.

### CODE REVIEW

File: client/containers/Users/Users.jsx [hunk 1]
```diff
@@ -46,8 +46,10 @@ class Users extends Component {
     this.props.fetchUsers('', false, page - 1);
   }
 
-  onSearch = (query) => {
-    this.props.fetchUsers(query);
+  onSearch = (query, filterBy) => {
+    if (query && query.length > 0) {
```
Comment (mostekcm) [inline client/containers/Users/Users.jsx:50]: What happens if you want to blank out the search bar?  Should we be searching for blank?

File: client/reducers/settings.js [hunk 2]
```diff
@@ -26,9 +26,14 @@ export const settings = createReducer(fromJS(initialState), { // eslint-disable-
...
           typeof field.display === 'string' &&
-          field.display.startsWith("function"))
-        // TODO: this is scary => what else can we do?
-          field.display = eval(`(${field.display})`);
+          field.display.startsWith('function')) {
+          try {
+            // TODO: this is scary => what else can we do?
+            field.display = eval(`(${field.display})`);
+          } catch (error) {
+            console.error(`The display function for field ${field.property} throws an error`, error);
```
Comment (mostekcm) [inline client/reducers/settings.js:34]: I wonder if this is exposing too much detail in the console...

File: server/routes/users.js [hunk 3]
```diff
@@ -49,19 +49,25 @@ export default (storage, scriptManager) => {
...
       .then((filter) => {
         const options = {
           sort: 'last_login:-1',
-          q: (req.query.search && filter) ? `(${req.query.search}) AND ${filter}` : req.query.search || filter,
+          q: (searchQuery && filter) ? `(${searchQuery}) AND ${filter}` : searchQuery || filter,
           per_page: req.query.per_page || 10,
           page: req.query.page || 0,
           include_totals: true,
-          fields: 'user_id,name,email,identities,picture,last_login,logins_count,multifactor,blocked,app_metadata',
+          fields: 'user_id,username,name,email,identities,picture,last_login,logins_count,multifactor,blocked,app_metadata',
```
Comment (mostekcm) [inline server/routes/users.js:70]: interesting.  I think we need to be building the list of fields from the list of userFields.  Some static default set, and then other fields based on the custom fields they are creating.  For example user_metadata isn't here, it probably should be, or should be dynamically added based on the custom fields set.

File: client/containers/Users/Users.jsx (hunk 1 above)
Comment (dctoon) [reply client/containers/Users/Users.jsx:50]: If you click the _reset_ button, a search with an empty query is performed.

File: client/reducers/settings.js (hunk 2 above)
Comment (dctoon) [reply client/reducers/settings.js:34]: Maybe, but the info is in the browsers memory anyway.

File: server/routes/users.js (hunk 3 above)
Comment (dctoon) [reply server/routes/users.js:70]: Indeed good catch, could this be part of the settings script?

**Pull request merged**